sdk/core/azure_core_amqp/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corp. All Rights Reserved.
+// cspell: words amqp
+
+#[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+use crate::fe2o3::error::Fe2o3AmqpError;
+
+pub enum ErrorKind {
+    #[cfg(all(feature = "enable-fe2o3-amqp", not(target_arch = "wasm32")))]
+    IronOxideError { source: Fe2o3AmqpError },
+    #[cfg(any(not(feature = "enable-fe2o3-amqp"), target_arch = "wasm32"))]
+    NoopError,
```

Comment:
I think this is unwieldy to use and maintain. The underlying implementation shouldn't be exposed like this, especially since it's varied on features. Be more generic. Can you uplift info from the fe2o3 crate if it's even useful? You can move or even ref data from underlying errors into a more generic name. The source itself is often a `Box<std::error::Error>` so anything implementing `Error` can do there.

Reply:
That was ... unpleasant, but I think I am happy with the results.